### ISSUE TITLE
Question about Maximum/Minimum value of the pixels.

### ISSUE THREAD

**Issue by chenmc1996** [louis2889184/adversarial_training#9]
Hi, thanks for the good work!
I'm tring to use the attack for my own purpose.
It seems that no normalization, such as  transforms.Normalize(), was used, so the input ranges from 0-1.
https://github.com/louis2889184/pytorch-adversarial-training/blob/1103fe300dc08f740b6870aebdd40a87d5690a45/cifar-10/main.py#L206-L210
As far as I know, it's comon pratice to normalize a tensor image with given mean and standard deviation. Then the input would have bigger range.
If so, when 
https://github.com/louis2889184/pytorch-adversarial-training/blob/1103fe300dc08f740b6870aebdd40a87d5690a45/cifar-10/src/attack/fast_gradient_sign_untargeted.py#L113
the perturbated images are total different. In my case, the attack destoryed the training process and the model went crazy actually.  
My question is what should I do to solve it? 
Besides, any other changes I should make if the attack is to be used for general purposes?

**Comment by ylsung** [comment]
You can replace `clamp_` by `torch.maximum` and `torch.minimum`, which can do the element-wise comparison between tensors. The sample code might be like

```python

transform_train = tv.transforms.Compose([ 
         tv.transforms.RandomCrop(32, padding=4, fill=0, padding_mode='constant'), 
         tv.transforms.RandomHorizontalFlip(), 
         tv.transforms.ToTensor(), 
         tv.transforms.Normalize(), # normalize by mean and std
     ]) 

min_tensor = torch.full((1, 3, 32, 32), 0)
max_tensor = torch.full((1, 3, 32, 32), 255)

min_tensor = transform_train(min_tensor)
max_tensor = transform_train(max_tensor)

def clamp(inputs, max, min):
    max = max.expand(inputs.shape[0], -1, -1, -1) # match the shape of max to be the same as which of inputs
    min = min.expand(inputs.shape[0], -1, -1, -1)
    
    return torch.maximum(torch.minimum(inputs, max), min)

# in fast_gradient_sign_untargeted.py
x = clamp(x, max_tensor, min_tensor)
```

The code snippet should fit in your needs.

**Comment by chenmc1996** [comment]
Thanks for your reply. It solves my question.
